error.vue

```diff
@@ -0,0 +1,27 @@
+<template>
+    <NuxtLayout>
+      <template #error>
+        <div class="nuxt-error">
+            <component :is="errorPage" :error="error" />
+        </div>
+      </template>
+    </NuxtLayout>
+  </template>
+  <script setup lang="ts">
+    import error404 from '~/pages/error/404.vue';
```

Comment:
This in my mind should be with an async component.  Nuxt3 makes it so hard.   Wait till you see my implementation, I am not crazy about mine.  Yours would be perfect is the components were async.